src/porepy/examples/example_params.py

```diff
@@ -0,0 +1,108 @@
+"""This file contains a complete set of top-level parameters for a PorePy model, along
+with a solver to run it. Use it as a starting point and copy the relevant parameters for
+your own problem.
+
+"""
+
+import porepy as pp
+from porepy.applications.material_values.fluid_values import (
+    extended_water_values_for_testing as water,
+)
+from porepy.applications.material_values.numerical_values import (
+    extended_numerical_values_for_testing as numerical_values,
+)
+from porepy.applications.material_values.reference_values import (
+    extended_reference_values_for_testing as reference_values,
+)
+from porepy.applications.material_values.solid_values import (
+    extended_granite_values_for_testing as granite,
+)
+from porepy.numerics.nonlinear.line_search import (
+    ConstraintLineSearch,
+    LineSearchNewtonSolver,
+    SplineInterpolationLineSearch,
+)
+
+
+# The most advanced nonlinear solver available (so far).
+class ConstraintLineSearchNonlinearSolver(
+    ConstraintLineSearch,
+    SplineInterpolationLineSearch,
+    LineSearchNewtonSolver,
+):
+    pass
+
+
+# Used for conversion of units.
+units = pp.Units()
+
+
+model_params = {
+    "linear_solver": "pypardiso",
+    "units": units,
+    "time_manager": pp.TimeManager(schedule=[0, 1], dt_init=1, constant_dt=True),
+    "reference_variable_values": pp.ReferenceVariableValues(**reference_values),  # type: ignore[arg-type]
+    "material_constants": {
+        "solid": pp.SolidConstants(**granite),  # type: ignore[arg-type]
+        "fluid": pp.FluidComponent(**water),  # type: ignore[arg-type]
+        "numerical": pp.NumericalConstants(**numerical_values),  # type: ignore[arg-type]
+    },
+    # Meshing
+    "grid_type": "cartesian",
+    "meshing_arguments": {
```

Comment:
Should we add a comment like "Depending on the type of grid, some subset of the below meshing arguments will be used."

Reply:
Added